tensorflow/python/kernel_tests/variable_scope_test.py

```diff
@@ -732,6 +732,115 @@ def testVarOpScopeNestedOuterScope(self):
           with ops.name_scope("scope2") as sc2:
             self.assertEqual(sc2, "outer_1/default/scope2/")
 
+  def testBasicWhenAuxiliaryNameScopeIsFalse(self):
+    with self.test_session():
+      with variable_scope.variable_scope("scope",
+                                         auxiliary_name_scope=False) as scope:
+        self.assertEqual(scope.original_name_scope, "")
```

Comment:
I think the `original_name_scope` is reasonable (but counterintuitive?) behavior. We almost never create `variable_scope` like this case. I'm afraid that user may misuse the argument and get lost in name scope.

Reply:
Perhaps it's better to restrict `auxiliary_name_scope` for reentering case.  Namely, `auxiliary_name_scope` only takes effect when VariableScope is passed.